app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -0,0 +1,53 @@
+package ar.com.wolox.android.example.ui.login;
+
+import android.util.Log;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import ar.com.wolox.android.example.utils.UserSession;
+import ar.com.wolox.wolmo.core.presenter.BasePresenter;
+
+/**
+ *
+ */
+public class LoginPresenter extends BasePresenter<ILoginView> {
+
+    private UserSession mUserSession;
+
+    @Inject
+    public LoginPresenter(UserSession mUserSession) {
+        this.mUserSession = mUserSession;
+    }
+
+    /**
+     *
+     * @param username Username
+     * @param password Password
+     */
+    public void startLogin(String username, String password) {
+        try {
+            if (this.mUserSession.getUsername() == null ||
+                    this.mUserSession.getPassword() == null) {
+                if (username.isEmpty() || password.isEmpty()) {
+                    getView().onEmptyForm();
```

Comment:
This is not bad, however, the decision to show the error message must be made from Presenter. We can send as parameters if user or password are empty and Fragment, will take both parameters and doesn't have to decide what to do.